pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,54 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (
+                isinstance(node.func, nodes.Name) and node.func.name in {"exit", "quit"}
+            )
+            or isinstance(node.parent, nodes.Lambda)
+        ):
+            return False
+
+        inferred = utils.safe_infer(node.func)
+
+        if (
+            not inferred
+            and isinstance(node.func, nodes.Attribute)
+            and node.func.attrname == "_exit"
+        ):
+            # edge case, try to detect os._exit
+            try:
```

Comment:
Let's just use the `try...except` for all of the inference. This is making two calls to `infer()` which are costly and I'm not sure this is cached correctly if `safe_infer` returns `Uninferable`.

Reply:
Just to make sure I understand your ask, you're saying to go from

```
        inferred = utils.safe_infer(node.func)

        if (
            not inferred
            and isinstance(node.func, nodes.Attribute)
            and node.func.attrname == "_exit"
        ):
            # edge case, try to detect os._exit
            try:
                inferred = next(node.func.infer())
            except (StopIteration, astroid.InferenceError):
                return False
            return inferred.qname() == "posix._exit"
```

to

```
        try:
            inferred = utils.safe_infer(node.func)
    
            if (
                not inferred
                and isinstance(node.func, nodes.Attribute)
                and node.func.attrname == "_exit"
            ):
                # edge case, try to detect os._exit
                inferred = next(node.func.infer())
                return inferred.qname() == "posix._exit"
        except Exception:  # pylint: disable=broad-except
            return False
```